utils/src/main/kotlin/FileMatcher.kt

```diff
@@ -43,18 +41,13 @@ class FileMatcher(
 
     constructor(vararg patterns: String) : this(patterns.asList())
 
-    private val matchers = patterns.map {
-        FileSystems.getDefault().getPathMatcher("glob:$it")
+    private val matcher = AntPathMatcher().apply {
+        setCachePatterns(true)
```

Comment:
Maybe better not do that as the docs say: "Default is for the cache to be on, but with the variant to automatically turn it off when encountering too many patterns to cache at runtime (the threshold is 65536), assuming that arbitrary permutations of patterns are coming in, with little chance for encountering a recurring pattern."

Reply:
I believe that it's ok to do that, because I think it's fine to assume that `val patterns: List<String>` will always contain less than 65536 entries. Do you agree?

Reply:
But if the only effect of setting this to true is that it will not stop caching at more than 65536 entries, and you don't expect more than this, why set it at all?

Reply:
For not creating a new instance of `StringPathMatcher` (internal of `AntPathMatcher`) on each call to `matches`.
Note: The implementation `as-is` creates instances only on the first call to `matches()`.